cmd/aws.go

```diff
@@ -98,10 +113,36 @@ func init() {
 	awsCmd.AddCommand(awsResourcesCmd)
 
 	// Required flags
-	awsCmd.Flags().String("access-key", "", "Access Key (required)")
-	awsCmd.Flags().String("secret-key", "", "Secret Key (required)")
-	awsCmd.Flags().String("region", "", "Region to search in, for now * it's not supported (required)")
+	awsCmd.Flags().String("aws-access-key", "", "Access Key (required)")
+	awsCmd.Flags().String("aws-secret-access-key", "", "Secret Key (required)")
+	awsCmd.Flags().String("aws-default-region", "", "Region to search in, for now * it's not supported (required)")
+	awsCmd.Flags().String("aws-shared-credentials-file", "", "Path to the AWS credential path")
+	awsCmd.Flags().String("aws-profile", "", "Name of the Profile to use with the Credentials")
 
 	// Filter flags
 	awsCmd.Flags().StringSliceVarP(&tags, "tags", "t", []string{}, "List of tags to filter with format 'NAME:VALUE'")
 }
+
+// loadAWSCredentials will first read from ENV and if AccessKey and SecretAccessKey are not found (both of them)
+// will fallback to the SharedCredentials with the profile
+func loadAWSCredentials() error {
+	creds := credentials.NewCredentials(&credentials.ChainProvider{
+		Providers: []credentials.Provider{
+			&credentials.EnvProvider{},
+			&credentials.SharedCredentialsProvider{Filename: viper.GetString("aws-shared-credentials-file"), Profile: viper.GetString("aws-profile")},
+		},
+	})
+
+	value, err := creds.Get()
+	if err != nil {
+		if awsE, ok := err.(awserr.Error); ok && awsE.Code() == "NoCredentialProviders" {
+			return nil
```

Comment:
Not sure why this error is ignored? Don't you end up without any credential in this situation?
Or it's just that the SharedCredentialProvider that fails? If so, it's still confusing as you commented: if ENV fails, fallback on Shared Credential Profile - so at this point it seems like there are no valid credentials found

Reply:
I added a comment to add more context to it (I planned to add it as it was also confusing for me haha).

> // The NoCredentialProviders is an error returned by Get to identify that none
// of the Providers (credentials.EnvProvider and credentials.SharedCredentialsProvider)
// did find any information.
// So we escape it means nothing was found by AWS

Basically it's an specific error saying that AWS did not find any credentials with the information provided, which we do not care as this is just a step to fill the credentials and after this step we do a validation of the required flags.

So we ignore this errors as it's confusing in this context and not needed as we only need the values if there are any defined.

Reply:
Hum, can the code run when going through the nil, or it simply will fail later on?
If it's gonna fail later on, I'd rather return a proper error here

Reply:
It'll fail later on as those values will be empty and we have a validation that the required ones are not empty.

Why return a proper error here? Which is the benefit? What is saying is "I did not find anything" why do we care? We'll always ignore it as it's not required for the user to use profile or env, they can still define things via flags which is out of this specific logic.

Reply:
OK, then it was my misunderstanding - if the code can still run later on - good for me